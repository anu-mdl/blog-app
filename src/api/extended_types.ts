import { CommentsRecord, PostsRecord, UsersRecord } from './api_types';

export type UserSocials = {
  twitter?: string;
  github?: string;
};

interface TOCItem {
  id: string;
  title: string;
  level: number;
}

// export type PostsRecordExtended = Omit<PostsRecord, 'author'> & {
//   tableOfContents?: TOCItem[];
//   author: UsersRecord<UserSocials>;
// };

export type PostsRecordExtended = PostsRecord & {
  tableOfContents?: TOCItem[];
  expand?: {
    author?: UsersRecord<UserSocials>;
  };
};

// export type PostsRecordExtended = Omit<PostsRecord, 'author'> & {
//   tableOfContents?: TOCItem[];
//   expand?: {
//     author: UsersRecord<UserSocials>;
//   };
// };

export type CommentsRecordExtended = Omit<CommentsRecord, 'author'> & {
  author: UsersRecord<UserSocials>;
};
