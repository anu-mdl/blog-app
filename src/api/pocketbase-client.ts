'use client';

import PocketBase from 'pocketbase';
import { TypedPocketBase } from './api_types';
import Cookies from 'js-cookie';

export function pocketbaseClient() {
  const pb = new PocketBase(process.env.NEXT_PUBLIC_PB_URL) as TypedPocketBase;
  pb.autoCancellation(false);

  const authCookie = Cookies.get('pb_auth_client');

  if (authCookie) {
    try {
      const authData = JSON.parse(decodeURIComponent(authCookie));
      pb.authStore.save(authData.token, authData.model);
    } catch (error) {
      console.error('Error parsing client auth cookie:', error);
    }
  }

  return pb;
}
