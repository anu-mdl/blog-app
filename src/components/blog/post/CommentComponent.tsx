import { CommentsRecordExtended } from '@/api/extended_types';
import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
import { formatDate } from '@/lib/utils';
import { Heart } from 'lucide-react';

export const CommentComponent = ({
  comment,
  isReply = false
}: {
  comment: CommentsRecordExtended;
  isReply?: boolean;
}) => (
  <div className={`space-y-3 ${isReply ? 'ml-12' : ''}`}>
    <div className="flex gap-3">
      <Avatar className="h-8 w-8 flex-shrink-0">
        <AvatarImage
          src={comment.author.avatar || '/image.svg'}
          alt={comment.author.username}
        />
        <AvatarFallback>{comment.author.username?.charAt(0)}</AvatarFallback>
      </Avatar>

      <div className="flex-1 min-w-0 space-y-2">
        <div className="flex items-center gap-2 text-sm">
          <span className="font-medium truncate">
            {comment.author.username}
          </span>
          <span className="text-muted-foreground whitespace-nowrap">
            {formatDate(comment.created || '')}
          </span>
        </div>

        <p className="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere">
          {comment.content}
        </p>

        <div className="flex items-center gap-4 text-xs text-muted-foreground">
          <button className="flex items-center gap-1 hover:text-foreground transition-colors">
            <Heart className="h-3 w-3" />
            <span>{comment.likes}</span>
          </button>
        </div>
      </div>
    </div>
  </div>
);
