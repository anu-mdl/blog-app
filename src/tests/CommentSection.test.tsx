import React from 'react';
import { render } from '@testing-library/react';
import { CommentComponent } from '@/components/blog/post/CommentComponent';
import { CommentsRecordExtended } from '@/api/extended_types';

jest.mock('@/lib/utils', () => ({
  formatDate: jest.fn().mockImplementation((date: string) => {
    if (date === '2025-06-12T10:30:00Z') return 'June 12, 2025';
    if (date === '2025-06-10T08:15:00Z') return 'June 10, 2025';
    return 'Mock Date';
  }),
  cn: jest
    .fn()
    .mockImplementation((...classes: (string | undefined)[]) =>
      classes.filter(Boolean).join(' ')
    )
}));

jest.mock('lucide-react', () => ({
  Heart: () => <svg data-testid="heart-icon" aria-label="heart" />
}));

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => <img {...props} />
}));

describe('CommentComponent', () => {
  const baseMockComment: CommentsRecordExtended = {
    id: 'comment-1',
    content: 'This is a test comment with some content to display.',
    created: '2025-06-12T10:30:00Z',
    likes: 5,
    author: {
      id: 'user-1',
      username: 'testuser',
      avatar: 'https://example.com/avatar.jpg',
      email: 'test@example.com',
      password: 'mockpassword',
      tokenKey: 'mocktoken'
    }
  };

  describe('Top-level comments', () => {
    it('renders a standard comment correctly', () => {
      const { container } = render(
        <CommentComponent comment={baseMockComment} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with high likes count', () => {
      const commentWithManyLikes = {
        ...baseMockComment,
        likes: 142
      };

      const { container } = render(
        <CommentComponent comment={commentWithManyLikes} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with zero likes', () => {
      const commentWithZeroLikes = {
        ...baseMockComment,
        likes: 0
      };

      const { container } = render(
        <CommentComponent comment={commentWithZeroLikes} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with long content', () => {
      const commentWithLongContent = {
        ...baseMockComment,
        content:
          'This is a very long comment that contains multiple sentences and should test how the component handles longer text content. It includes various punctuation marks, and tests the word wrapping functionality of the component. This helps ensure that the layout remains consistent even with extensive content.'
      };

      const { container } = render(
        <CommentComponent comment={commentWithLongContent} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with multiline content', () => {
      const commentWithMultilineContent = {
        ...baseMockComment,
        content:
          'First line of the comment.\n\nSecond paragraph after line break.\n\nThird paragraph with more content.'
      };

      const { container } = render(
        <CommentComponent comment={commentWithMultilineContent} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Reply comments', () => {
    it('renders reply comment with indentation', () => {
      const { container } = render(
        <CommentComponent comment={baseMockComment} isReply={true} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders reply comment with different author', () => {
      const replyComment = {
        ...baseMockComment,
        id: 'comment-reply-1',
        content: 'This is a reply to the original comment.',
        author: {
          ...baseMockComment.author,
          id: 'user-2',
          username: 'replyuser',
          avatar: 'https://example.com/reply-avatar.jpg'
        }
      };

      const { container } = render(
        <CommentComponent comment={replyComment} isReply={true} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Avatar handling', () => {
    it('renders comment without avatar (fallback)', () => {
      const commentWithoutAvatar = {
        ...baseMockComment,
        author: {
          ...baseMockComment.author,
          avatar: undefined
        }
      };

      const { container } = render(
        <CommentComponent comment={commentWithoutAvatar} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with empty avatar string', () => {
      const commentWithEmptyAvatar = {
        ...baseMockComment,
        author: {
          ...baseMockComment.author,
          avatar: ''
        }
      };

      const { container } = render(
        <CommentComponent comment={commentWithEmptyAvatar} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with single character username', () => {
      const commentWithSingleCharUsername = {
        ...baseMockComment,
        author: {
          ...baseMockComment.author,
          username: 'A',
          avatar: undefined
        }
      };

      const { container } = render(
        <CommentComponent comment={commentWithSingleCharUsername} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with long username', () => {
      const commentWithLongUsername = {
        ...baseMockComment,
        author: {
          ...baseMockComment.author,
          username: 'verylongusernamethatmighttruncate'
        }
      };

      const { container } = render(
        <CommentComponent comment={commentWithLongUsername} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Date handling', () => {
    it('renders comment with different date', () => {
      const commentWithDifferentDate = {
        ...baseMockComment,
        created: '2025-06-10T08:15:00Z'
      };

      const { container } = render(
        <CommentComponent comment={commentWithDifferentDate} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with empty date', () => {
      const commentWithEmptyDate = {
        ...baseMockComment,
        created: ''
      };

      const { container } = render(
        <CommentComponent comment={commentWithEmptyDate} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with undefined date', () => {
      const commentWithUndefinedDate = {
        ...baseMockComment,
        created: undefined
      };

      const { container } = render(
        <CommentComponent comment={commentWithUndefinedDate} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Edge cases', () => {
    it('renders comment with special characters in content', () => {
      const commentWithSpecialChars = {
        ...baseMockComment,
        content:
          'Comment with special chars: <>&"\'`~!@#$%^&*()_+-=[]{}|;:,.<>?'
      };

      const { container } = render(
        <CommentComponent comment={commentWithSpecialChars} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with emoji in content', () => {
      const commentWithEmoji = {
        ...baseMockComment,
        content:
          'Great post! üëç Really enjoyed reading this üòä Keep up the good work! üöÄ'
      };

      const { container } = render(
        <CommentComponent comment={commentWithEmoji} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders comment with undefined username', () => {
      const commentWithUndefinedUsername = {
        ...baseMockComment,
        author: {
          ...baseMockComment.author,
          username: undefined
        }
      };

      const { container } = render(
        <CommentComponent comment={commentWithUndefinedUsername} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
