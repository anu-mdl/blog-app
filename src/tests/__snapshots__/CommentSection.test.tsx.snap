// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentComponent Avatar handling renders comment with empty avatar string 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Avatar handling renders comment with long username 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        v
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          verylongusernamethatmighttruncate
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Avatar handling renders comment with single character username 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        A
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          A
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Avatar handling renders comment without avatar (fallback) 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Date handling renders comment with different date 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 10, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Date handling renders comment with empty date 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          Mock Date
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Date handling renders comment with undefined date 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          Mock Date
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Edge cases renders comment with emoji in content 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        Great post! 👍 Really enjoyed reading this 😊 Keep up the good work! 🚀
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Edge cases renders comment with special characters in content 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        Comment with special chars: &lt;&gt;&"'\`~!@#$%^&*()_+-=[]{}|;:,.&lt;&gt;?
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Edge cases renders comment with undefined username 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      />
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        />
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Reply comments renders reply comment with different author 1`] = `
<div
  class="space-y-3 ml-12"
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        r
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          replyuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a reply to the original comment.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Reply comments renders reply comment with indentation 1`] = `
<div
  class="space-y-3 ml-12"
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Top-level comments renders a standard comment correctly 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Top-level comments renders comment with high likes count 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            142
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Top-level comments renders comment with long content 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a very long comment that contains multiple sentences and should test how the component handles longer text content. It includes various punctuation marks, and tests the word wrapping functionality of the component. This helps ensure that the layout remains consistent even with extensive content.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Top-level comments renders comment with multiline content 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        First line of the comment.

Second paragraph after line break.

Third paragraph with more content.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            5
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentComponent Top-level comments renders comment with zero likes 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex gap-3"
  >
    <span
      class="relative flex size-8 shrink-0 overflow-hidden rounded-full h-8 w-8 flex-shrink-0"
      data-slot="avatar"
    >
      <span
        class="bg-muted flex size-full items-center justify-center rounded-full"
        data-slot="avatar-fallback"
      >
        t
      </span>
    </span>
    <div
      class="flex-1 min-w-0 space-y-2"
    >
      <div
        class="flex items-center gap-2 text-sm"
      >
        <span
          class="font-medium truncate"
        >
          testuser
        </span>
        <span
          class="text-muted-foreground whitespace-nowrap"
        >
          June 12, 2025
        </span>
      </div>
      <p
        class="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere"
      >
        This is a test comment with some content to display.
      </p>
      <div
        class="flex items-center gap-4 text-xs text-muted-foreground"
      >
        <button
          class="flex items-center gap-1 hover:text-foreground transition-colors"
        >
          <svg
            aria-label="heart"
            data-testid="heart-icon"
          />
          <span>
            0
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
